maven-jxr-plugin/src/it/settings.xml

```diff
@@ -60,18 +46,10 @@ under the License.
             <enabled>true</enabled>
           </snapshots>
         </pluginRepository>
-        <pluginRepository>
-          <id>apache.snapshots</id>
-          <name>Apache Snapshot Repository</name>
-          <url>http://repository.apache.org/content/groups/snapshots/</url>
-          <releases>
-            <enabled>false</enabled>
-          </releases>
-          <snapshots>
-            <enabled>true</enabled>
-          </snapshots>
-        </pluginRepository>        
       </pluginRepositories>
     </profile>
   </profiles>
+  <activeProfiles>
+    <activeProfile>it-repo</activeProfile>
+  </activeProfiles>
 </settings>
```

Comment:
Why do we need this settings file at all? I ran the tests without and they success. There is no settings file in MPIR.

Reply:
Needed for m-invoker-p https://maven.apache.org/plugins/maven-invoker-plugin/examples/fast-use.html

Reply:
Interesting, in other plugins I haven't seen it. Did you check the time difference actually?

Reply:
Wow, right, but not used: https://github.com/apache/maven-project-info-reports-plugin/blob/a91b480cf52a152f1afbe52f691577aaf3f54866/pom.xml#L591-L611

Reply:
One thing is time consumed be test and second is how many items are download from remote repository.

without `settings.xml`

```
[INFO] --- maven-invoker-plugin:3.3.0:run (integration-test) @ maven-jxr-plugin ---
[INFO] Building: simple-project/pom.xml
[INFO] run post-build script verify.groovy
[INFO]           simple-project/pom.xml ........................... SUCCESS (23.1 s)
[INFO] Building: JXR-100_parameterlink/pom.xml
[INFO] run post-build script verify.groovy
[INFO]           JXR-100_parameterlink/pom.xml .................... SUCCESS (3.5 s)
[INFO] Building: aggregate/pom.xml
[INFO] run post-build script verify.groovy
[INFO]           aggregate/pom.xml ................................ SUCCESS (10.3 s)
[INFO] Building: mixed-plugin-report/pom.xml
[INFO] run post-build script verify.groovy
[INFO]           mixed-plugin-report/pom.xml ...................... SUCCESS (13.0 s)
[INFO] Building: JXR-135_innerclasses/pom.xml
[INFO]           JXR-135_innerclasses/pom.xml ..................... SUCCESS (1.8 s)
[INFO] Building: JXR-143_nofork/pom.xml
[INFO] run post-build script verify.groovy
[INFO]           JXR-143_nofork/pom.xml ........................... SUCCESS (5.5 s)
[INFO] Building: new-site-config/pom.xml
[INFO] run post-build script verify.groovy
[INFO]           new-site-config/pom.xml .......................... SUCCESS (5.3 s)
[INFO] -------------------------------------------------
[INFO] Build Summary:
[INFO]   Passed: 7, Failed: 0, Errors: 0, Skipped: 0
[INFO] -------------------------------------------------
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for Maven JXR Parent 3.2.1-SNAPSHOT:
[INFO] 
[INFO] Maven JXR Parent ................................... SUCCESS [  2.545 s]
[INFO] Maven JXR .......................................... SUCCESS [  3.217 s]
[INFO] Maven JXR Plugin ................................... SUCCESS [01:10 min]
[INFO] ------------------------------------------------------------------------
```

with `settings.xml`

```
[INFO] --- maven-invoker-plugin:3.3.0:run (integration-test) @ maven-jxr-plugin ---
[INFO] Building: simple-project/pom.xml
[INFO] run post-build script verify.groovy
[INFO]           simple-project/pom.xml ........................... SUCCESS (5.9 s)
[INFO] Building: JXR-100_parameterlink/pom.xml
[INFO] run post-build script verify.groovy
[INFO]           JXR-100_parameterlink/pom.xml .................... SUCCESS (1.8 s)
[INFO] Building: aggregate/pom.xml
[INFO] run post-build script verify.groovy
[INFO]           aggregate/pom.xml ................................ SUCCESS (6.1 s)
[INFO] Building: mixed-plugin-report/pom.xml
[INFO] run post-build script verify.groovy
[INFO]           mixed-plugin-report/pom.xml ...................... SUCCESS (12.2 s)
[INFO] Building: JXR-135_innerclasses/pom.xml
[INFO]           JXR-135_innerclasses/pom.xml ..................... SUCCESS (1.6 s)
[INFO] Building: JXR-143_nofork/pom.xml
[INFO] run post-build script verify.groovy
[INFO]           JXR-143_nofork/pom.xml ........................... SUCCESS (3.5 s)
[INFO] Building: new-site-config/pom.xml
[INFO] run post-build script verify.groovy
[INFO]           new-site-config/pom.xml .......................... SUCCESS (4.2 s)
[INFO] -------------------------------------------------
[INFO] Build Summary:
[INFO]   Passed: 7, Failed: 0, Errors: 0, Skipped: 0
[INFO] -------------------------------------------------
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for Maven JXR Parent 3.2.1-SNAPSHOT:
[INFO] 
[INFO] Maven JXR Parent ................................... SUCCESS [  2.145 s]
[INFO] Maven JXR .......................................... SUCCESS [  2.473 s]
[INFO] Maven JXR Plugin ................................... SUCCESS [ 42.544 s]
[INFO] ------------------------------------------------------------------------
```

Reply:
I will add do MIRP 😄

Reply:
hehe parent magic 😄 

https://github.com/apache/maven-parent/blob/550a810a356ca4479717771417cd9dea7b09ed8c/maven-plugins/pom.xml#L235

so MPIR use settings.xml in IT tests

Reply:
Oh man, I was blind for years. Can we apply this here as well?